### ISSUE TITLE
transformer_relative decoding fails: "Key transformer/transformer/... not found" 

### ISSUE THREAD

**Issue by martinpopel** [tensorflow/tensor2tensor#674]
I train `HPARAMS=transformer_relative_big` ([Self-Attention with Relative Position Representations](https://arxiv.org/abs/1803.02155)) and it goes OK,
but the decoding fails with
```
NotFoundError (see above for traceback): Key transformer/transformer/body/decoder/layer_0/encdec_attention/layer_prepostprocess/layer_norm/layer_norm_bias not found in checkpoint
         [[Node: save/RestoreV2_1 = RestoreV2[dtypes=[DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, ..., DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT, DT_FLOAT], _device="/job:localhost/replica:0/task:0/device:CPU:
0"](_arg_save/Const_0_0, save/RestoreV2_1/tensor_names, save/RestoreV2_1/shape_and_slices)]]
         [[Node: save/RestoreV2_1/_55 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnati
on=1, tensor_name="edge_62_save/RestoreV2_1", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]
```
The checkpoint contains a key `transformer/body/decoder/layer_0/encdec_attention/layer_prepostprocess/layer_norm/layer_norm_bias`
but the decoding is looking for a key starting with **transformer/transformer**, i.e. there is one extra "transformer" name space.
The key is the first of decoder keys, so I guess it is not important that it is layer_norm_bias.

Interestingly, when I decode with HPARAMS=transformer_big_single_gpu (instead of the correct HPARAMS=transformer_relative_big), the decoding does not fail, it just produces very low BLEU, as could be expected because the model was not trained to get input absolute-position embeddings and at the same time it is missing the relative-position embeddings.

I made sure I am using the same T2T version and hparams for training and decoding.
tensor2tensor==1.5.6 (yes, it is not released yet, I am using it from #668)
tensorflow==1.6.0

**Comment by martinpopel** [comment]
Oh, fixing the bug was faster than writing this issue:-) (but I got the idea while writing the issue).
This issue will be automatically closed once #675 is merged.

**Comment by songmeixu** [comment]
Till v1.9.0, this error still appears, for other people encounter this problem, here is my reason and solution:

Reason:
as @martinpopel said, 
> "the decoding is looking for a key starting with transformer/transformer"
and i construct a custom transformer model, and modified its name as:
```python
  def __init__(self, *args, **kwargs):
    super(TransformerScorer, self).__init__(*args, **kwargs)
    self._name = "transformer_my_custom_name"
    self._base_name = "transformer"
```

solution:
DO NOT modify these name...
